packetbeat/flows/worker.go

```diff
@@ -96,17 +174,73 @@ func newFlowsWorker(
 	return makeWorker(processor, tickDuration, ticksTimeout, ticksPeriod, 10)
 }
 
-func makeWorker(
-	processor *flowsProcessor,
-	tickDuration time.Duration,
-	ticksTimeout, ticksPeriod int,
-	align int64,
-) (*worker, error) {
+// gcd returns the greatest common divisor of a and b.
+func gcd(a, b int64) int64 {
+	// FIXME: This can be rewritten as:
```

Comment:
Justification here: https://play.golang.com/p/mL4wWPfQmGn

Note that the contract is not identical for the two function, but under the domain of use here (`a` and `b` are never negative), they are equivalent.

Reply:
The rewrite seems quite a bit simpler. I would go with it.

Reply:
Yes, it was recently devised by Euclid.

I intend to go through and send separate PRs for all the TODOs here to avoid conflating logic changes and clean up.